pysat/tests/test_meta.py

```diff
@@ -16,1384 +15,2389 @@
+        self.frame_list = list(
+            self.testInst.meta['profiles']['children'].keys())
+        for dvar in self.frame_list:
+            if dvar == 'density':
+                ho_meta[dvar] = {'barrels': 'A'}
+            else:
+                ho_meta[dvar] = {'Monkeys': 'are fun', 'bananas': 2}
+
+        # The 'units', 'desc' and other labels used on self.testInst are
+        # applied to the input metadata to ensure everything remains
+        # consistent across the object.
+        self.testInst['profiles'] = {'data': self.testInst.data['profiles'],
+                                     'units': 'V', 'long_name': 'The Doors',
+                                     'meta': ho_meta}
+        self.meta = self.testInst.meta
+
+        # Test the nD metadata
+        assert self.testInst.meta['profiles', 'long_name'] == 'The Doors'
+        testing.assert_list_contains(self.frame_list,
+                                     self.meta.ho_data['profiles'])
+        testing.assert_list_contains(self.frame_list,
+                                     self.meta['profiles']['children'])
+
+        for label in ['units', 'desc']:
+            assert self.meta['profiles']['children'].hasattr_case_neutral(label)
+
+        assert self.meta['profiles']['children']['density', 'units'] == 'A'
+        assert self.meta['profiles']['children']['density', 'desc'] == ''
+
+        for dvar in ['dummy_str', 'dummy_ustr']:
+            assert self.meta['profiles']['children'][dvar, 'desc'] == 'are fun'
+            assert self.meta['profiles']['children'][dvar, 'bananas'] == 2
+        return
+
+    # TODO(#789): remove tests for higher order meta
+    def test_concat_w_ho(self):
+        """Test `meta.concat` adds new meta objects with higher order data."""
+
+        # Create meta data to concatenate
+        meta2 = pysat.Meta()
+        meta2['new3'] = {'units': 'hey3', 'long_name': 'crew_brew'}
+        meta2['new4'] = pysat.Meta(pds.DataFrame({
+            'units': {'new41': 'hey4'}, 'long_name': {'new41': 'crew_brew'},
+            'bob_level': {'new41': 'max'}}))
+
+        # Perform and test for successful concatenation
+        self.meta = self.meta.concat(meta2)
+        assert self.meta['new3'].units == 'hey3'
+        assert self.meta['new4'].children['new41'].units == 'hey4'
+        return
+
+    # TODO(#913): remove tests for 2d metadata
+    def test_concat_not_strict_w_ho_collision(self):
+        """Test non-strict concat with overlapping higher-order data."""
+
+        # Set the meta object
+        self.set_meta(inst_kwargs={'platform': 'pysat', 'name': 'testing2d'})
+
+        # Create a second object with the same higher-order data variables
+        concat_meta = pysat.Meta()
+        for dvar in self.meta.keys_nD():
+            concat_meta[dvar] = self.meta[dvar]
+
+            # Change the units of the HO data variables
+            for cvar in concat_meta[dvar].children.keys():
+                # HERE FIX, DOESN'T WORK
```

Comment:
That one is your comment 😸  Quick check shows that the setting "UpdatedUnits" doesn't take. Further debugging shows that we even assign the new inputs in the code, 
```
                        if good_set:
                            self._data.loc[var, ikey] = to_be_set
                        print(self._data.loc[var])
```
confirmed via the print. For reasons unknown, those changes are no longer available once the code exits. Perhaps the changes are applied to a copy of the child meta?

```
In [5]: inst.meta["series_profiles"].children["series_profiles"] = {"units": "newUnits"}

units                    newUnits
long_name         series_profiles
notes                            
desc         Testing series data.
value_min                     0.0
value_max                     inf
fill                          NaN
Name: series_profiles, dtype: object

In [6]: inst.meta["series_profiles"].children["series_profiles"]
Out[6]: 
units                         m/s
long_name         series_profiles
notes                            
desc         Testing series data.
value_min                     0.0
value_max                     inf
fill                          NaN
children                     None
Name: series_profiles, dtype: object
```

Looks like that is the case.
Changes below in `_meta.py` sort assignment issue out. There is a warning above the code in _meta.py that copies are returned to avoid 'setting with copy warning`. Testing with this push if we still get that issue.
```
                meta_row = self.data.loc[new_key] #.copy()
                if new_key in self.keys_nD():
                    meta_row.at['children'] = self.ho_data[new_key] #.copy()
                else:
                    meta_row.at['children'] = None  # Return empty meta instance
```